pysmt/solvers/z3.py

```diff
@@ -128,21 +143,61 @@ def solve(self, assumptions=None):
         sres = str(res)
         assert sres in ['unknown', 'sat', 'unsat']
         if sres == 'unknown':
-            raise SolverReturnedUnknownResultError()
+            raise SolverReturnedUnknownResultError
+        return (sres == 'sat')
+
+    def get_unsat_core(self):
+        """After a call to solve() yielding UNSAT, returns the unsat core as a
+        set of formulae"""
+        return self.get_named_unsat_core().values()
+
+    def _named_assertions_map(self):
+        if self.options.unsat_cores_mode is not None:
+            return {t[0] : (t[1],t[2]) for t in self.assertions}
+        return None
+
+    def get_named_unsat_core(self):
+        """After a call to solve() yielding UNSAT, returns the unsat core as a
+        dict of names to formulae"""
+        if self.options.unsat_cores_mode is None:
```

Comment:
ok, but I expect that for any UnsatCoreSolver you will be interested in running this check, i.e., verify that the solver is in a valid state. This might be a symptom that the current approach of extending solver interfaces is not good enough. Lets leave it like this, for now.